sycl/test-e2e/NonUniformGroups/ballot_group_algorithms.cpp

```diff
@@ -0,0 +1,131 @@
+  sycl::buffer<bool, 1> NoneBuf{sycl::range{32}};
+  sycl::buffer<bool, 1> ReduceBuf{sycl::range{32}};
+  sycl::buffer<bool, 1> ExScanBuf{sycl::range{32}};
+  sycl::buffer<bool, 1> IncScanBuf{sycl::range{32}};
+
+  const auto NDR = sycl::nd_range<1>{32, 32};
+  Q.submit([&](sycl::handler &CGH) {
+    sycl::accessor TmpAcc{TmpBuf, CGH, sycl::write_only};
+    sycl::accessor BarrierAcc{BarrierBuf, CGH, sycl::write_only};
+    sycl::accessor BroadcastAcc{BroadcastBuf, CGH, sycl::write_only};
+    sycl::accessor AnyAcc{AnyBuf, CGH, sycl::write_only};
+    sycl::accessor AllAcc{AllBuf, CGH, sycl::write_only};
+    sycl::accessor NoneAcc{NoneBuf, CGH, sycl::write_only};
+    sycl::accessor ReduceAcc{ReduceBuf, CGH, sycl::write_only};
+    sycl::accessor ExScanAcc{ExScanBuf, CGH, sycl::write_only};
+    sycl::accessor IncScanAcc{IncScanBuf, CGH, sycl::write_only};
+    const auto KernelFunc =
+        [=](sycl::nd_item<1> item) [[sycl::reqd_sub_group_size(32)]] {
+          auto WI = item.get_global_id();
+          auto SG = item.get_sub_group();
+
+          // Split into odd and even work-items
+          bool Predicate = WI % 2 == 0;
+          auto BallotGroup = syclex::get_ballot_group(SG, Predicate);
+
+          // Check all other members' writes are visible after a barrier
+          TmpAcc[WI] = 1;
+          sycl::group_barrier(BallotGroup);
+          size_t Visible = 0;
+          for (size_t Other = 0; Other < 32; ++Other) {
+            if (WI % 2 == Other % 2) {
+              Visible += TmpAcc[Other];
+            }
+          }
+          BarrierAcc[WI] = Visible;
+
+          // Simple check of group algorithms
+          uint32_t OriginalLID = SG.get_local_linear_id();
+          uint32_t LID = BallotGroup.get_local_linear_id();
+
+          uint32_t BroadcastResult =
+              sycl::group_broadcast(BallotGroup, OriginalLID, 0);
+          if (Predicate) {
+            BroadcastAcc[WI] = (BroadcastResult == 0);
+          } else {
+            BroadcastAcc[WI] = (BroadcastResult == 1);
+          }
+
+          bool AnyResult = sycl::any_of_group(BallotGroup, Predicate);
+          if (Predicate) {
+            AnyAcc[WI] = (AnyResult == true);
+          } else {
+            AnyAcc[WI] = (AnyResult == false);
+          }
+
+          bool AllResult = sycl::all_of_group(BallotGroup, Predicate);
+          if (Predicate) {
+            AllAcc[WI] = (AllResult == true);
+          } else {
+            AllAcc[WI] = (AllResult == false);
+          }
+
+          bool NoneResult = sycl::none_of_group(BallotGroup, Predicate);
+          if (Predicate) {
+            NoneAcc[WI] = (NoneResult == false);
+          } else {
+            NoneAcc[WI] = (NoneResult == true);
+          }
+
+          uint32_t ReduceResult =
+              sycl::reduce_over_group(BallotGroup, 1, sycl::plus<>());
+          ReduceAcc[WI] =
+              (ReduceResult == BallotGroup.get_local_linear_range());
```

Comment:
Strictly speaking, the previous test only verified `get_local_range()` and not `get_local_linear_range()` but we can leave this to CTS.